### ISSUE TITLE
cannot authenticate with github, completely stuck. help :(

### ISSUE THREAD

**Issue by sw-yx** [FreeCodeCamp/FreeCodeCamp#11809]
Challenge [Save your Code Revisions Forever with Git](https://www.freecodecamp.com/challenges/save-your-code-revisions-forever-with-git) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.98 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

Hi folks, i really really need help here as this seems to be a fatal error (fatal as in i have no idea how i can proceed learning until i resolve this). I have followed the steps of the tutorial closely until the "git push origin master" stage. when i reached here, the system asked me to log in, and then does not recognize my credentials: 

```
swyx:~/workspace/hello-world (master) $ git push origin master
Username for 'https://github.com': sw-yx
Password for 'https://sw-yx@github.com': 
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/sw-yx/hello-world.git/
```

Here is what i have done to check that i am not going mad:
- logged in on browser to verify i know my own password
- changed my password to verify i know my own password
- the guys on the chat (https://gitter.im/FreeCodeCamp/HelpBackEnd?at=58384f7916207f7b0ea478e3) made me do git remote show origin which i did

```
swyx:~/workspace/hello-world (master) $ git remote show origin
remote origin
Fetch URL: https://github.com/sw-yx/hello-world.git
Push URL: https://github.com/sw-yx/hello-world.git
HEAD branch: (unknown)
swyx:~/workspace/hello-world (master) $
```

basically i am extremely extremely frustrated that I'm dying at such a simple step. this isnt even the hard part! Timo assures me that this is not because i screewed up somethign in the ssh setup. please help me out. I will happily answer any and all diagnostic questions, I really, really really don't want to stop because of such a stupid issue.

**Comment by sw-yx** [comment]
obviously if i am facing this issue we might want to modify the instructions to help other people who may encounter this same thing.

**Comment by sw-yx** [comment]
this was fix for my issue. thanks to @systimotic 

http://stackoverflow.com/questions/17659206/git-push-results-in-authentication-failed
